Sources/eppo/dto/EppoValue.swift

```diff
@@ -152,21 +152,52 @@ public class EppoValue : Decodable, Equatable {
         return value;
     }
 
-    public func toHashedString() -> String {
-        // todo: this is a bit of a hack. we should probably just use a switch
-        // to handle the different types.
-        var str = ""
-        if let value = self.stringValue {
-            str = value
-        } else if let array = self.stringArrayValue {
-            str = array.joined(separator: " ,")
+    public func toEppoString() -> String {
+        switch self.type {
+        case .Boolean:
+            return try! self.getBoolValue() ? "true" : "false"
+        case .Numeric:
+            if let doubleVal = try? self.getDoubleValue() {
+                if floor(doubleVal) == doubleVal {
+                    // If the double value is an integer, return as integer string
+                    return String(format: "%.0f", doubleVal)
+                } else {
+                    // Otherwise, return the double as is
+                    return String(doubleVal)
+                }
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
+        case .String:
+            // todo: should this throw?
+            // it should not since we checked the type is a string above
+            // but to be true to the type system we should throw here
+            if let value = try? self.getStringValue() {
+                return value
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
+        case .ArrayOfStrings:
+            if let value = try? self.getStringArrayValue() {
+                return value.joined(separator: " ,")
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
+        case .Null:
+            return "null"
         }
+    }
 
+    public func toHashedString() -> String {
```

Comment:
this function is used for the assignment logging to keep a consistent cache entry size (https://github.com/Eppo-exp/eppo-ios-sdk/blob/main/Sources/eppo/AssignmentCache.swift#L30). it's probably extraneous and I could just use the new function in this PR. Thoughts welcome.

Reply:
Not related to this PR, but wouldn't it make more sense to use the variation key instead of the variation value when hashing? Practically speaking wouldn't make a difference, but we avoid having to hash json blobs

Reply:
Yea good point I will review my old PR for why I decided to use this.